flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphDeploymentTest.java

```diff
@@ -182,14 +179,20 @@ public void testBuildDeploymentDescriptor() {
 			ExecutionJobVertex ejv = eg.getAllVertices().get(jid2);
 			ExecutionVertex vertex = ejv.getTaskVertices()[3];
 
-			ExecutionGraphTestUtils.SimpleActorGatewayWithTDD instanceGateway =
-				new ExecutionGraphTestUtils.SimpleActorGatewayWithTDD(
-					TestingUtils.directExecutionContext(),
-					blobCache);
+			SimpleAckingTaskManagerGateway taskManagerGateway = new SimpleAckingTaskManagerGateway();
+			CompletableFuture<TaskDeploymentDescriptor> tdd = new CompletableFuture<>();
+			taskManagerGateway.setSubmitConsumer(taskDeploymentDescriptor -> {
+				try {
+					taskDeploymentDescriptor.loadBigData(blobCache);
+				} catch (Exception e) {
+					e.printStackTrace();
```

Comment:
Alternatively we could use a `ThrowingConsumer` here and actually catch exceptions in the `SimpleAckingTaskManagerGateway`, that logs them and returns a failed future.

Throwing an erro here is essentially undefined behavior.